ktesting/testinglogger.go

```diff
@@ -94,6 +105,25 @@ func NewLogger(t TL, c *Config) logr.Logger {
 	})
 }
 
+// Stop ensures that all future log entries get discarded. This is necessary
```

Comment:
Discarding output will make it hard to see that goroutines have leaked and debug why they are still running.

I think a better solution will be to redirect output to the global klog logger with a unique prefix. Something like

`<test>: leaked goroutine: <message>`

Reply:
Also, this entire new API call is unnecessary.

There is `testing.T.Cleanup`, so ktesting itself can ensure that it gets notified when the test is done.

/hold

Reply:
/hold cancel

Using testing.T.Cleanup implies extending out TL interface, which is an API change. ktesting is still experimental, so I think it is okay and worth it.

Reply:
I found a way around the API break: instead of requiring `Name` and `Cleanup`, the constructor now only uses them if implemented.